src/ser/mod.rs

```diff
@@ -635,15 +649,15 @@ impl<'a> ser::SerializeStruct for &'a mut Serializer {
         self.output += key;
         self.output += ":";
 
-        if self.is_pretty() {
+        if self.indentation_enabled() {
```

Comment:
I don't think this check is correct

Reply:
`self.indentation_enabled()` and `self.is_pretty()` are equivalent, the former is just clearer so I removed `self.is_pretty`.

Ah, and it shouldn't check for indentation here! So this means there was a mistake before, which is that we should check if `self.pretty` is some, not if indentation is enabled (so it turns out I was confused by the naming myself...).

Reply:
Hmm, turns out we simply don't use any spaces right now if the depth limit was reached.